Heroesprofile api update

Astraeus- opened the pull request:
Closes #153 In my testing with a subset of our user-base this handled their rate-limit. I increased the `set_time_limit` to 60 to be on the safe side to not let the script time out, as they reset the permitted requests every minute.

We'll need to update our authorization token, as we were required to generate a new one. I can send this to you when you deploy.

Fabian-Sommer commented on plugins/rikki/heroeslounge/classes/hotslogs/IDFetcher.php:
```diff
@@ -11,8 +11,12 @@ public static function fetchIDs()
         foreach ($sloths as $sloth) {
-            set_time_limit(30);
-            IDFetcher::fetchIDHeroesProfile($sloth);
+            set_time_limit(60);
```
Please assume that our server might just ignore these

Fabian-Sommer commented on plugins/rikki/heroeslounge/classes/hotslogs/IDFetcher.php:
```diff
@@ -11,8 +11,12 @@ public static function fetchIDs()
+                sleep($throttleTime);
+            }
         }
     }
```
This function is probably not actually needed anymore

Astraeus- replied:
Will update this in a different PR for #154

Fabian-Sommer commented on plugins/rikki/heroeslounge/classes/hotslogs/IDFetcher.php:
```diff
@@ -55,22 +59,41 @@ public static function fetchIDHeroesProfile($sloth)
-                if (array_key_exists("blizz_id", $data[0])) {
-                    $sloth->heroesprofile_id = $data[0]["blizz_id"];
+                if (array_key_exists("blizz_id", $data)) {
+                    $sloth->heroesprofile_id = $data["blizz_id"];
```
I assume that these are changes in their API

Astraeus- replied:
That's correct, they changed this in their response

Fabian-Sommer commented on plugins/rikki/heroeslounge/classes/mmr/MMRFetcher.php:
```diff
@@ -21,8 +25,12 @@ public static function updateMMRs()
+
+            if ($throttleTime > 0) {
+                sleep($throttleTime);
+            }
```
We will have to see if this works on our server. I had bad experiences with increasing time limit in the past

Astraeus- commented:
I need to re-think the throttle logic a bit for this, as I realized that in the current state it'll drop the request that it failed on instead of retrying it after our limit resets.

Astraeus- commented:
I tested the following with a large number of requests and didn't run into any timeout issues myself even at 30, so I changed it back to that. I added retry logic for the throttled request as well.
